### PR TITLE
Register incr/decr calls with the timeout statistics

### PR DESCRIPTION
**Pull request by eabbott**
Synchronous incr/decr calls will now update the internal statistics spymemcached uses to determine a remote memcached node is timing out, and thus unavailable. This change is particularly aimed at JavaRateLimiter classes that do not use the regular memcached pool, but instead use the @Rehashable pool. With nothing else using the @Rehashable pool, a timing out memcached node would never be pulled from the active duty and would not only time out continuously, but would never fulfill the @Reshashable trait of failing over to an available node. 

This failure condition is only for memcached nodes that continuously time out, such as our most common memcached failure situation in which an AWS instance just disappears. Connections to that server continually time out until a developer forces the instance to reboot or replaces the instance.

And to further clarify spymemcached driver behavior, the internal timeout statistics are a different mechanism of failure detection than the simple 'connection refused' situation such as a memcached process stopping/crashing. Situations in which the network layer knows the process is down (ie connection refused) currently act as desired.

@jdwyah Here is your patch as requested.
@HiJon89  @axiak  @sit you all helped PR the last few changes I'd made to this codebase. This particular change is a bit less intrusive than the last couple. Thanks!

### CODE REVIEW

**Comment by eabbott** [comment]
The synchronous mutate method does not create an operation that extends the OperationFuture that other calls use which properly registers timeouts. This PR mimics the behavior of the OperationFuture get() method.

https://github.com/HubSpot/java-memcached-client/blob/2.7.3-hs/src/main/java/net/spy/memcached/internal/OperationFuture.java#L64

File: src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java [hunk 1]
```diff
@@ -44,8 +44,8 @@
 	private ArrayList<Operation> reconnectBlocked;
 	private long defaultOpTimeout;
 
-	// operation Future.get timeout counter
-	private Object continuousTimeoutLock = new Object();
+	// operation Future.get (and mutate) timeout counter
+	private final Object continuousTimeoutLock = new Object();
```
Comment (sit) [inline src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java:48]: Is this change relevant to the correctness of the change? Should it be made separately?

**Comment by sit** [comment]
Looks reasonable. I would copy the nice PR description/comments you wrote and put them in the commit message itself.

File: src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java (hunk 1 above)
Comment (eabbott) [inline src/main/java/net/spy/memcached/protocol/TCPMemcachedNodeImpl.java:48]: The comment change belongs with this change. The addition of final could be a separate commit.

**Comment by eabbott** [comment]
Splitting this into two commits, and including the comments in the commit. It'll be easiest to create a new PR, closing this one out.
